tests/test_experiment_server.py

```diff
@@ -248,23 +248,22 @@ def test_records_notification_if_debug_mode(self, a, webapp):
         assert models.Notification.query.one().event_type == u'AssignmentSubmitted'
 
     def test_records_notification_if_bot_recruiter(self, a, webapp, active_config):
-        active_config.extend({'recruiter': u'bots'})
         webapp.get('/worker_complete?participant_id={}'.format(
-            a.participant().id)
+            a.participant(recruiter_id='bots').id)
         )
         assert models.Notification.query.one().event_type == u'BotAssignmentSubmitted'
 
     def test_records_no_notification_mturk_recruiter_and_nondebug(self, a, webapp, active_config):
         active_config.extend({'mode': u'sandbox'})
         webapp.get('/worker_complete?participant_id={}'.format(
-            a.participant().id)
+            a.participant(recruiter_id='mturk').id)
         )
         assert models.Notification.query.all() == []
 
     def test_records_notification_for_non_mturk_recruiter(self, a, webapp, active_config):
         active_config.extend({'mode': u'sandbox', 'recruiter': u'CLIRecruiter'})
```

Comment:
In some cases you no longer specify a recruiter in the config, and rely on the Participant `recruiter` attribute, but not always. This is a little confusing. Can the participant attr be used exclusively here?

Reply:
I didn't review or change the tests that weren't failing. That's by design, to try to keep backwards-compatibility.